readinessProbe/readiness/server_test.go

```diff
@@ -0,0 +1,98 @@
+package readiness
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "github.com/openshift/ocs-osd-deployer/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("ManagedOCS Readiness Probe", func() {
+	// Define utility constants for object names and testing timeouts/durations and intervals.
+	const (
+		ManagedOCSName = "test-managedocs"
+		TestNamespace  = "default"
+
+		timeout  = time.Second * 10
+		duration = time.Second * 10
+		interval = time.Millisecond * 250
+	)
+
+	getResourceKey := func(obj runtime.Object) client.ObjectKey {
+		key, err := client.ObjectKeyFromObject(obj)
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		return key
+	}
+
+	When("managedocs reports its components as \"not ready\"", func() {
+		BeforeEach(func() {
+			ctx := context.Background()
+
+			managedOCS := &v1.ManagedOCS{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      ManagedOCSName,
+					Namespace: TestNamespace,
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, managedOCS)).Should(Succeed())
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, getResourceKey(managedOCS), managedOCS)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+
+			managedOCS.Status.Components.StorageCluster.State = v1.ComponentPending
+
+			Expect(k8sClient.Status().Update(ctx, managedOCS)).Should(Succeed())
+		})
+
+		It("should cause the readiness logic to report \"not ready\"", func() {
+			managedOCSResource := types.NamespacedName{
+				Name:      ManagedOCSName,
+				Namespace: TestNamespace,
+			}
+			ready, err := isReady(k8sClient, managedOCSResource)
```

Comment:
I would prefer to test an actual response from the HTTP server (with an HTTP code) and not function invocation.
Calling the function directly you lose coverage on all surrounding code (including the code governing the returned HTTP status)